plugins/de.cognicrypt.codegenerator.tests/src/de/cognicrypt/codegenerator/generator/test/QuestionsToTemplateTests.java

```diff
@@ -0,0 +1,544 @@
+package de.cognicrypt.codegenerator.generator.test;
+
+import static org.junit.Assert.assertEquals;
+
+
+import org.junit.Test;
+
+import de.cognicrypt.codegenerator.question.Answer;
+import de.cognicrypt.codegenerator.question.Question;
+import de.cognicrypt.codegenerator.tasks.Task;
+import de.cognicrypt.codegenerator.testutilities.TestUtils;
+import de.cognicrypt.codegenerator.wizard.AltConfigWizard;
+import java.util.HashMap;
+
+/**
+ * The JUnit Plug-in tests check the correctness of CogniCrypt's code generation feature
+ * in determining the template path when the user chooses an arbitrary task and a set of 
+ * given question-answers in the wizard dialog. This is important since the code generation
+ * depends on the computed path to generate the proper template for a given input from
```

Comment:
This raises a bigger worrying concern for me. I understand this test is important since the given template is used elsewhere. But shouldnt the part where it is used just directly call this "template path" function instead of relying on hard coding the path there. It is possible that I have just misunderstood how this path is used elsewhere. So my review is mostly a request for clarification on how this works. Thanks.

Reply:
These JUnit tests only check that the expected template path is computed when a user opens the wizard and chooses an arbitrary task together with a set of questions-answers (if there are any). The template path is only used at the end of the wizard (at the Locator page) when the user clicks 'Generate', to generate the template code into the file chosen by the user. 
It might also be possible to check the constructed template path on the last page of the wizard, but the test cases would need to include every wizard page call and each wizard page type would need to include many changes, which would make the existing code a little complicated. Therefore, a simpler approach is used for testing whether the correct template path is computed, where a wizard is created in every test case with a given task and set of Q-A. About the hardcoded path, maybe you were confused about a typo of mine which is now fixed, where I had messed up the order of the actual and expected arguments in the assertEquals() method call. The hardcoded path is the expected result of the test and the other argument is the path that the wizard computed with the given arguments. Hope I have clarified your doubts :)

Reply:
Yes. Thanks :)